src/storage/src/filter_key_extractor.rs

```diff
@@ -376,15 +375,60 @@ impl FilterKeyExtractorManager {
     /// Acquire a `MultiFilterKeyExtractor` by `table_id_set`
     /// Internally, try to get all `filter_key_extractor` from `hashmap`. Will block the caller if
     /// `table_id` does not util version update (notify), and retry to get
+    async fn acquire(&self, table_id_set: HashSet<u32>) -> HummockResult<FilterKeyExtractorImpl> {
+        self.inner.acquire(table_id_set).await
+    }
+}
+
+#[derive(Clone)]
+pub enum FilterKeyExtractorManager {
+    RpcFilterKeyExtractorManager(Arc<RpcFilterKeyExtractorManager>),
+    StaticFilterKeyExtractorManager(StaticFilterKeyExtractorManager),
```

Comment:
We should also wrap `StaticFilterKeyExtractorManager` in Arc. Otherwise, the clone will be come heavy since the whole HashMap with all table catalogs will be copied.

Reply:
Nice catch!